.github/workflows/ci.yaml

```diff
@@ -0,0 +1,31 @@
+name: CI
+
+on:
+  - pull_request
+  - push
+
+concurrency:
+  group: ${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: Ruby
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - 2.7
+          - 3.0
+          - 3.1
```

Comment:
You should always quote the versions. Otherwise it may end up being interpreted as floats. So `3.0` becomes `3` which other tooling can interpret as "the latest Ruby 3".